packages/beeq/src/components/alert/bq-alert.tsx

```diff
@@ -125,31 +133,44 @@ export class BqAlert {
   /** Method to be called to hide the alert component */
   @Method()
   async hide(): Promise<void> {
-    this.handleHide();
+    await this.handleHide();
   }
   /** Method to be called to show the alert component */
   @Method()
   async show(): Promise<void> {
-    this.handleShow();
+    await this.handleShow();
   }
 
   // Local methods
   // Internal business logic.
   // These methods cannot be called from the host element.
   // =======================================================
 
-  private handleHide = () => {
+  private handleHide = async () => {
     const ev = this.bqHide.emit(this.el);
     if (!ev.defaultPrevented) {
+      await leave(this.alertElement);
       this.open = false;
+      this.handleTransitionEnd();
     }
   };
```

Comment:
Let's apply the `.is-hidden` class in the Host via JS, right after the leave animation, that way we guarantee that the animation runs properly.

```suggestion
  private handleHide = async () => {
    const ev = this.bqHide.emit(this.el);
    if (!ev.defaultPrevented) {
      await leave(this.alertElement);
      this.el.classList.add('is-hidden');
      this.handleTransitionEnd();
      this.open = false;
    }
  };
```